src/Razor/src/Microsoft.VisualStudio.Editor.Razor/LegacyProjectEngineFactory_3_0.cs

```diff
@@ -23,8 +25,13 @@ public RazorProjectEngine Create(RazorConfiguration configuration, RazorProjectF
 
             return RazorProjectEngine.Create(configuration, fileSystem, b =>
             {
+                CompilerFeatures.Register(b);
```

Comment:
This adds component discovery. It has to be a separate gesture because of the assembly boundary. I don't like it either :(